src/main/resources/com/google/api/codegen/js/sample.snip

```diff
@@ -0,0 +1,181 @@
+  function makeApiCall() {
+    @if class.needParams
+      var {@class.paramsVarName} = {
+        @join field : class.fields if class.fields on BREAK.add(BREAK)
+          @if field.description
+            @join line : util.getDocLines(field.description, 100)
+              // {@line}
+            @end
+
+
+          @end
+          {@field.name}: {@field.defaultValue},  // TODO: Update placeholder value.
+          @if field.example
+            // {@field.example}
+          @end
+        @end 
+      };
+
+    @end
+    @if class.hasRequestBody
+      var {@class.requestBodyVarName} = {
+        @switch class.methodVerb
+        @case "PATCH"
+          // TODO: Add desired properties to the request body. Only these properties
+          // will be changed.
+        @case "PUT"
+          // TODO: Add desired properties to the request body. All existing properties
+          // will be replaced.
+        @default
+          // TODO: Add desired properties to the request body.
+        @end
+      };
+
+    @end
+    @if class.hasMediaUpload
+      // TODO: To do media uploads, it's more appropriate to use CORS. See
+      // https://developers.google.com/api-client-library/javascript/features/cors
+
+    @end
+    @if class.hasMediaDownload
+      // TODO: To do media downloads, it's more appropriate to use CORS. See
+      // https://developers.google.com/api-client-library/javascript/features/cors
+
+    @end
+    @if class.isPageStreaming
+      {@pageStreamingFuncSignature(class)};
+    @else
+      var {@class.requestVarName} = {@requestFuncSignature(class)};
+      @if class.hasResponse
+        {@class.requestVarName}.then(function({@class.responseVarName}) {
+          // TODO: Change code below to process the `{@class.responseVarName}` object:
+          console.log({@class.responseVarName}.result);
+        }, function(reason) {
+          console.log('error: ' + reason.result.error.message);
+        });
+      @else
+        {@class.requestVarName}.then(function() {}, function(reason) {
+          console.log('error: ' + reason.result.error.message');
+        });
+      @end
+    @end
+  }
+
+  @if class.isPageStreaming
+    function {@pageStreamingFuncSignature(class)} {
+      var {@class.requestVarName} = {@requestFuncSignature(class)};
+      request.then(function({@class.responseVarName}) {
+        var {@class.pageStreaming.pageVarName} = {@class.responseVarName}.result.{@class.pageStreaming.resourceFieldName};
+        @if class.pageStreaming.isResourceRepeated
+          @if class.pageStreaming.isResourceMap
+            Object.keys({@class.pageStreaming.pageVarName}).forEach(function({@class.pageStreaming.resourceKeyVarName}) {
+              // TODO: Change code below to process each `{@class.pageStreaming.resourceKeyVarName}` property:
+              console.log({@class.pageStreaming.resourceKeyVarName} + ': ' + JSON.stringify({@class.pageStreaming.pageVarName}[{@class.pageStreaming.resourceKeyVarName}], null, 2));
```

Comment:
I believe JSON is always available with Node.js, but is that true for all Javascript environments we want to support?